packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -1130,6 +1130,47 @@ void main() {
     }
   });
 
+  testWidgets('Enable filtering with custom filter callback that filter text case sensitive', (WidgetTester tester) async {
+    final ThemeData themeData = ThemeData();
+    final TextEditingController controller = TextEditingController();
+    addTearDown(controller.dispose);
+
+    await tester.pumpWidget(MaterialApp(
+      theme: themeData,
+      home: Scaffold(
+        body: DropdownMenu<TestMenu>(
+          requestFocusOnTap: true,
+          enableFilter: true,
+          filterCallback: (List<DropdownMenuEntry<TestMenu>> entries, String filter) {
+            return entries.where((DropdownMenuEntry<TestMenu> element) => element.label.contains(filter)).toList();
+          },
+          dropdownMenuEntries: menuChildren,
+          controller: controller,
+        ),
+      ),
+    ));
+
+    // Open the menu
+    await tester.tap(find.byType(DropdownMenu<TestMenu>));
+    await tester.pump();
+
+    await tester.enterText(find.byType(TextField).first, 'item');
+    expect(controller.text, 'item');
+    await tester.pumpAndSettle();
+    for (final TestMenu menu in TestMenu.values) {
+      expect(find.widgetWithText(MenuItemButton, menu.label).hitTestable(), findsNothing);
+    }
+
+    await tester.enterText(find.byType(TextField).first, 'Item');
+    expect(controller.text, 'Item');
+    await tester.pumpAndSettle();
+    expect(find.widgetWithText(MenuItemButton, 'Item 0').hitTestable(), findsOneWidget);
+    expect(find.widgetWithText(MenuItemButton, 'Item 2').hitTestable(), findsOneWidget);
```

Comment:
I think we should be looking to see that an item isn't there to see that it's filtered out. Can we check that 'Menu 1' isn't there, or update the filter and see if some of these disappear?

Reply:
Oh that's a good point! Missed this! @dacianf Could you add one more line to check `Menu 1`?
```suggestion
    expect(find.widgetWithText(MenuItemButton, 'Menu 1').hitTestable(), findsNothing);
    expect(find.widgetWithText(MenuItemButton, 'Item 2').hitTestable(), findsOneWidget);
```